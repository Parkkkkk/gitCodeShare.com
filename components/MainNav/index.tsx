import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';
import {
  styled,
  GithubButton,
  GithubBadge,
  SmDownloadButton,
  MainNavMenu,
  SContainer,
  FlexRightBox,
  RowFlex,
  SLogoImage,
} from '../../styledComponents';
import { Layout } from 'antd';
import Link from 'next/link';
import { IAppStore } from '../../stores/AppStore';

interface IProps {
  appStore?: IAppStore;
  className?: string;
}

const { Header } = Layout;
const Nav = styled(Header as any)`
  ${RowFlex};
  align-items: center;
  height: 70px;
  background-color: ${props => props.theme.primaryColor};
`;

const NavContainer = styled(SContainer as any)`
  display: inherit;
  align-items: center;

  a {
    margin: 0;

    &:hover,
    &:focus,
    &:active,
    &:visited {
      text-decoration: none;
    }
  }

  h1 {
    margin: 0 20px 0 0;
  }

  [data-align] {
    display: inherit;
    align-items: center;
  }

  [data-violet] {
    color: ${props => props.theme.colorPalette.blueViolet};
    &.ant-badge {
      background-color: ${props => props.theme.colorPalette.blueViolet};
    }
  }

  [data-magenta] {
    color: ${props => props.theme.colorPalette.deepMagenta};
    &.ant-badge {
      background-color: ${props => props.theme.colorPalette.deepMagenta};
    }
  }
`;

const RightBox = styled(FlexRightBox as any)`
  & {
    button {
      margin-right: 30px;
    }
    [data-align] {
      margin-right: 15px;
    }
  }
`;

@inject('appStore')
@observer
class MainNav extends Component<IProps> {
  render() {
    const { className, appStore } = this.props;
    const { fork, star } = appStore;
    return (
      <Nav className={className}>
        <NavContainer>
          <h1 data-align>
            <Link href="/">
              <a data-align>
                <SLogoImage />
              </a>
            </Link>
          </h1>
          <MainNavMenu mode="horizontal">
            <MainNavMenu.Item>
              <Link href="/contributors">
                <a>Contributors</a>
              </Link>
            </MainNavMenu.Item>
            <MainNavMenu.Item>
              <Link href="/features">
                <a>Features</a>
              </Link>
            </MainNavMenu.Item>
          </MainNavMenu>
          <RightBox>
            <Link href="/extension">
              <a
                href="//chrome.google.com/webstore/detail/gitcodeshare/fegiblhnedcljeapaigmgnfjceochhhg"
                target="blank"
                data-align
              >
                <SmDownloadButton icon="chrome">Add to Chrome</SmDownloadButton>
              </a>
            </Link>
            <a href="https://github.com/kosslab-kr/gitCodeShare.com/stargazers" target="blank" data-align>
              <GithubButton data-violet icon="github">
                Star
              </GithubButton>
              <GithubBadge data-violet count={star} />
            </a>
            <a href="https://github.com/kosslab-kr/gitCodeShare.com/network/members" target="blank" data-align>
              <GithubButton data-magenta icon="github">
                Fork
              </GithubButton>
              <GithubBadge data-magenta count={fork} />
            </a>
          </RightBox>
        </NavContainer>
      </Nav>
    );
  }
}

export default MainNav;
