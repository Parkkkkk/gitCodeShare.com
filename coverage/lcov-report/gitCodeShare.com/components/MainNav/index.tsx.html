<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for gitCodeShare.com/components/MainNav/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">gitCodeShare.com/components/MainNav</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">63.16% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">63.16% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/19</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import {
  styled,
  GithubButton,
  GithubBadge,
  SmDownloadButton,
  MainNavMenu,
  SContainer,
  FlexRightBox,
  RowFlex,
} from '../../styledComponents';
import { Layout } from 'antd';
import Link from 'next/link';
import { ApiProvider } from '../../providers';
&nbsp;
interface IProps {
  className?: string;
}
&nbsp;
const { Header } = Layout;
const Nav = styled(Header as any)`
  ${RowFlex};
  align-items: center;
  height: 70px;
  background-color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >props.theme.primaryColor}</span>;
`;
&nbsp;
const NavContainer = styled(SContainer as any)`
  display: inherit;
  align-items: center;
&nbsp;
  h1,
  a {
    margin: 0;
  }
&nbsp;
  [data-align] {
    display: inherit;
    align-items: center;
  }
&nbsp;
  [data-violet] {
    color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >props.theme.colorPalette.blueViolet}</span>;
    &amp;.ant-badge {
      background-color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >props.theme.colorPalette.blueViolet}</span>;
    }
  }
&nbsp;
  [data-magenta] {
    color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >props.theme.colorPalette.deepMagenta}</span>;
    &amp;.ant-badge {
      background-color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >props.theme.colorPalette.deepMagenta}</span>;
    }
  }
`;
&nbsp;
const RightBox = styled(FlexRightBox as any)`
  &amp; {
    button {
      margin-right: 30px;
    }
    [data-align] {
      margin-right: 15px;
    }
  }
`;
&nbsp;
class MainNav extends Component&lt;IProps&gt; {
  state = {
    fork: 0,
    star: 0,
  };
&nbsp;
  componentDidMount() {
    this.getData();
  }
&nbsp;
  async getData() {
    try {
      const fork = await ApiProvider.GithubRequest.getForkNum();
      const star = <span class="cstat-no" title="statement not covered" >await ApiProvider.GithubRequest.getStarNum();</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ fork, star });</span>
    } catch (err) {
      this.setState({ fork: 0, star: 0 });
      console.log(err);
    }
  }
&nbsp;
  render() {
    const { className } = this.props;
    const { fork, star } = this.state;
    return (
      &lt;Nav className={className}&gt;
        &lt;NavContainer&gt;
          &lt;h1 data-align&gt;
            &lt;Link href="/"&gt;
              &lt;a data-align&gt;
                &lt;img src="/static/logo.png" alt="GitShareCode" /&gt;
              &lt;/a&gt;
            &lt;/Link&gt;
          &lt;/h1&gt;
          &lt;MainNavMenu mode="horizontal"&gt;
            &lt;MainNavMenu.Item&gt;
              &lt;Link href="/contributors"&gt;
                &lt;a&gt;Contributors&lt;/a&gt;
              &lt;/Link&gt;
            &lt;/MainNavMenu.Item&gt;
            &lt;MainNavMenu.Item&gt;
              &lt;Link href="/features"&gt;
                &lt;a&gt;Features&lt;/a&gt;
              &lt;/Link&gt;
            &lt;/MainNavMenu.Item&gt;
          &lt;/MainNavMenu&gt;
          &lt;RightBox&gt;
            &lt;Link href="/extension"&gt;
              &lt;a data-align&gt;
                &lt;SmDownloadButton icon="chrome"&gt;Add to Chrome&lt;/SmDownloadButton&gt;
              &lt;/a&gt;
            &lt;/Link&gt;
            &lt;a href="https://github.com/kosslab-kr/gitCodeShare.com/stargazers" target="blank" data-align&gt;
              &lt;GithubButton data-violet icon="github"&gt;
                Star
              &lt;/GithubButton&gt;
              &lt;GithubBadge data-violet count={star} /&gt;
            &lt;/a&gt;
            &lt;a href="https://github.com/kosslab-kr/gitCodeShare.com/network/members" target="blank" data-align&gt;
              &lt;GithubButton data-magenta icon="github"&gt;
                Fork
              &lt;/GithubButton&gt;
              &lt;GithubBadge data-magenta count={fork} /&gt;
            &lt;/a&gt;
          &lt;/RightBox&gt;
        &lt;/NavContainer&gt;
      &lt;/Nav&gt;
    );
  }
}
&nbsp;
export default MainNav;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Oct 20 2018 17:56:07 GMT+0900 (KST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
