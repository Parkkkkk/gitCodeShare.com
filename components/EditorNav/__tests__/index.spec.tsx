import React from 'react';
import { shallow } from 'enzyme';
import MainNav from '../';

describe('MainNav', () => {
  it('should render', () => {
    const tree = shallow(<MainNav />);
    expect(tree).toMatchSnapshot();
  });
});
