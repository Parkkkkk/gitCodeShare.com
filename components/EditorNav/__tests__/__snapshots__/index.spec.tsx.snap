// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditorNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <h1
          data-align={true}
        >
          <Link
            href="/"
          >
            <a
              data-align={true}
            >
              <SLogoImage />
            </a>
          </Link>
        </h1>
      </ForwardRef>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <h1
          data-align={true}
        >
          <Link
            href="/"
          >
            <a
              data-align={true}
            >
              <SLogoImage />
            </a>
          </Link>
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            href="/"
          >
            <a
              data-align={true}
            >
              <SLogoImage />
            </a>
          </Link>,
          "data-align": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <a
              data-align={true}
            >
              <SLogoImage />
            </a>,
            "href": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <SLogoImage />,
              "data-align": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
          "type": [Function],
        },
        "type": "h1",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "EditorNav__NavContainer-sc-5nwyvu-1",
          "isStatic": true,
          "rules": Array [
            "width:100%;max-width:1200px;margin:0 auto;",
            "display:inherit;align-items:center;h1,a{margin:0;}[data-align]{display:inherit;align-items:center;}",
          ],
        },
        "displayName": "EditorNav__NavContainer",
        "foldedComponentIds": Array [
          "StyledLayout__SContainer-sc-1ec2pa7-2",
        ],
        "render": [Function],
        "styledComponentId": "EditorNav__NavContainer-sc-5nwyvu-1",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "EditorNav__Nav-sc-5nwyvu-0",
        "isStatic": false,
        "rules": Array [
          "
    position: relative;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: flex-start;
    flex-wrap: wrap;
    min-width: auto;
    width: auto;
    min-height: auto;
    height: auto;
    ;
  ",
          ";align-items:center;height:70px;background-color:",
          [Function],
          ";",
        ],
      },
      "displayName": "EditorNav__Nav",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "EditorNav__Nav-sc-5nwyvu-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <h1
            data-align={true}
          >
            <Link
              href="/"
            >
              <a
                data-align={true}
              >
                <SLogoImage />
              </a>
            </Link>
          </h1>
        </ForwardRef>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h1
            data-align={true}
          >
            <Link
              href="/"
            >
              <a
                data-align={true}
              >
                <SLogoImage />
              </a>
            </Link>
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              href="/"
            >
              <a
                data-align={true}
              >
                <SLogoImage />
              </a>
            </Link>,
            "data-align": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                data-align={true}
              >
                <SLogoImage />
              </a>,
              "href": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <SLogoImage />,
                "data-align": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
            "type": [Function],
          },
          "type": "h1",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "EditorNav__NavContainer-sc-5nwyvu-1",
            "isStatic": true,
            "rules": Array [
              "width:100%;max-width:1200px;margin:0 auto;",
              "display:inherit;align-items:center;h1,a{margin:0;}[data-align]{display:inherit;align-items:center;}",
            ],
          },
          "displayName": "EditorNav__NavContainer",
          "foldedComponentIds": Array [
            "StyledLayout__SContainer-sc-1ec2pa7-2",
          ],
          "render": [Function],
          "styledComponentId": "EditorNav__NavContainer-sc-5nwyvu-1",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "EditorNav__Nav-sc-5nwyvu-0",
          "isStatic": false,
          "rules": Array [
            "
    position: relative;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: flex-start;
    flex-wrap: wrap;
    min-width: auto;
    width: auto;
    min-height: auto;
    height: auto;
    ;
  ",
            ";align-items:center;height:70px;background-color:",
            [Function],
            ";",
          ],
        },
        "displayName": "EditorNav__Nav",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "EditorNav__Nav-sc-5nwyvu-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
